main.go

```diff
@@ -1,54 +1,67 @@
 package main
 
 import (
+	"bufio"
 	"context"
+	"fmt"
 	"os"
-	"os/signal"
+	"time"
 
-	"github.com/ONSdigital/dp-search-data-extractor/service"
+	kafka "github.com/ONSdigital/dp-kafka/v2"
+	"github.com/ONSdigital/dp-search-data-extractor/config"
+	"github.com/ONSdigital/dp-search-data-extractor/models"
+	"github.com/ONSdigital/dp-search-data-extractor/schema"
 	"github.com/ONSdigital/log.go/v2/log"
-	"github.com/pkg/errors"
 )
 
 const serviceName = "dp-search-data-extractor"
 
-var (
-	// BuildTime represents the time in which the service was built
-	BuildTime string
-	// GitCommit represents the commit (SHA-1) hash of the service that is running
-	GitCommit string
-	// Version represents the version of the service that is running
-	Version string
-)
-
 func main() {
 	log.Namespace = serviceName
 	ctx := context.Background()
-
-	if err := run(ctx); err != nil {
-		log.Fatal(ctx, "fatal runtime error", err)
+	// Get Config
+	config, err := config.Get()
+	if err != nil {
+		log.Fatal(ctx, "error getting config", err)
+		os.Exit(1)
+	}
+	// Create Kafka Producer
+	pChannels := kafka.CreateProducerChannels()
```

Comment:
The producer should be created in the service package, I saw that you also create one there, which is ok. Can you undo all the changes in this `main.go` file, please? Otherwise the other components wouldn't be created, and the service wouldn't work.

We use the following structure:
- `main.go` just starts the service package, and does a single os.Exit(1) if there is any error returned by the service
- `service/service.go` has a Service struct which contains all the dependencies required by the service, and a Run() func that initialises and starts all dependencies.

Reply:
Undo done. This makes sense as I too was confused while creating the producer. Thanks.